libcloud/dns/drivers/dnspod.py

```diff
@@ -0,0 +1,345 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sys
+
+
+from libcloud.dns.types import Provider, ZoneDoesNotExistError, \
+    ZoneAlreadyExistsError, RecordDoesNotExistError, RecordAlreadyExistsError
+from libcloud.dns.base import DNSDriver, Zone, Record, RecordType
+from libcloud.common.dnspod import DNSPodConnection, DNSPodResponse, \
+    DNSPodException
+from libcloud.utils.py3 import urlencode
+
+__all__ = [
+    'DNSPodDNSDriver'
+]
+
+
+class DNSPodDNSResponse(DNSPodResponse):
+    pass
+
+
+class DNSPodDNSConnection(DNSPodConnection):
+    responseCls = DNSPodDNSResponse
+
+
+class DNSPodDNSDriver(DNSDriver):
+    name = 'DNSPod'
+    website = 'https://dnspod.com'
+    type = Provider.DNSPOD
+    connectionCls = DNSPodDNSConnection
+
+    RECORD_TYPE_MAP = {
+        RecordType.A: 'A',
+        RecordType.AAAA: 'AAAA',
+        RecordType.CNAME: 'CNAME',
+        RecordType.MX: 'MX',
+        RecordType.NS: 'NS',
+        RecordType.PTR: 'PTR',
+        RecordType.SOA: 'SOA',
+        RecordType.SRV: 'SRV',
+        RecordType.TXT: 'TXT'
+    }
+
+    def list_zones(self):
+        action = '/Domain.List'
+        data = {'user_token': self.key, 'format': 'json'}
```

Comment:
@Kami hey man any updated on this? I have fixed the other problems, this is the only one left.